dev/NavigationView/NavigationView.cpp

```diff
@@ -1925,6 +1936,11 @@ void NavigationView::ChangeSelection(const winrt::IInspectable& prevItem, const
         // And SelectionChanged is after we selected the new item.
         UnselectPrevItem(prevItem, nextItem);
         ChangeSelectStatusForItem(nextItem, true /*selected*/);
+        if (m_shouldRaiseItemInvokedAfterSelection)
+        {
+            RaiseItemInvokedForNavigationViewItem(nextItem.try_as<winrt::NavigationViewItem>());
```

Comment:
`nextItem` is not always guaranteed to be a `NavigationViewItem`. It could be some custom data type defined by the user in databinding scenarios.
Use the `NavigationViewItemOrSettingsContentFromData` call instead if you want to retrieve a container from an item.

Also, instead of calling `RaiseItemInvokedForNavigationViewItem` function here, it might make more sense to call `RaiseItemInvoked` directly since you already have access to all the required input parameters such as `recommendedDirection` .

Reply:
Okay, switched to using `NavigationViewItemOrSettingsContentFromData` and calling RaiseItemInvoked now.